components/authentication/login.jsx

```diff
@@ -1,163 +1,144 @@
-import React from 'react';
-import { StyleSheet, Text, View, TextInput, Image } from 'react-native';
-import {Button, Provider } from 'react-native-paper'
+import * as React from 'react';
+import {
+  SafeAreaView, View, StyleSheet, Image, TextInput,
+} from 'react-native';
+import { Button } from 'react-native-paper';
+import firebase from 'firebase';
 
-export default class login extends React.Component {
-    state={
-        email:"",
-        password:"",
-        showPassword:true,
-    }
+export default ({ navigation }) => {
+  const [email, setEmail] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  const [showPassword, setShowPassword] = React.useState(true);
 
-    loginPressed = () => {
-        alert("Login button pressed")
-    }
+  function loginPress() {
+    firebase.auth().signInWithEmailAndPassword(email, password)
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMsg = error.message;
+        switch (errorCode) {
+          case 'auth/wrong-password':
+            alert('Incorrect password.');
+            console.log('Incorrect password.');
+            break;
 
-    forgotPressed = () => {
-        alert("forgot password pressed")
-    }
+          default:
+            console.log(`error logging in: ${errorMsg}`);
+        }
+      });
+  }
 
-    signUpPressed = () => {
-        alert("sign up pressed")
-    }
+  function signupPress() {
+    navigation.navigate('Signup');
+    console.log("Navigating to signup...")
```

Comment:
Display message to user if they have an incorrect password.

Reply:
This part is just if they don't have an account yet and navigate to the signup page, so no info is saved